python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -208,6 +213,81 @@ async def prepare_for_flat(self, config_name: str) -> None:
             task_setup_electrometer,
         )
 
+    async def calculate_optimized_exposure_times(
+        self, wavelengths: list, config_data: dict
+    ) -> list:
+        """Calculates the exposure times for the fiber spectrograph and
+        the electrometer given camera exposure times and other
+        relevant configuration paramaters.
+
+        This includes wavelength dependent information for the fiber
+        spectrograph and also includes the logic for the size of
+        the electrometer buffer.
+
+        Parameters
+        ----------
+        wavelengths : `list`
+            List of all wavelengths for this exposure list
+        config_data : `dict`
+            All information from configuration file
+
+        Returns
+        -------
+        exposure_list : `array`
+            List of exposure information, including wavelength and camera,
+            fiberspectrograph and electrometer exposure times.
+
+        """
+        exposures = []
+        for wavelength in wavelengths:
+            for exptime in config_data["exposure_times"]:
+                if config_data["use_electrometer"]:
+                    time_sep = (
+                        config_data["electrometer_integration_time"] * 3.07
```

Comment:
Where are these hard coded numbers coming from? Do they have any meaning? I would suggest either making them local variables with names expressing what they are or making them class attributes. 

The same comment applies to the 3 numbers that appears here, e.g. `3.07`, `0.00254` and `16667`.